eng/pipelines/templates/steps/generate-project-list.yml

```diff
@@ -1,31 +1,28 @@
 # This job template uses the artifact list in the ci.yml file
 # to generate a variable at build time that can be passed into
 # the Maven -pl command-line switch to constrain what subset of
-# projects are targetted based on the artifacts this pipeline
+# projects are targeted based on the artifacts this pipeline
 # produces. The -am and -amd switches are then applied as
 # appropriate to constrain the full graph that is built depending
 # on whether we are doing a recorded test play back or a full live
 # test.
 
 steps:
+- template: ../steps/cache-maven-repository.yml
+
 - pwsh: |
-    Write-Host "##vso[task.setvariable variable=ProjectList;]"
-  displayName: Initialize project list variable
+    $artifacts = '${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json
```

Comment:
@benbp starting to use this convertToJson trick to eliminate a bunch of noise in build steps that aren't necessary.